examples/games/flappy_bevy.rs

```diff
@@ -0,0 +1,278 @@
+    mut spawn_bird_events: EventReader<SpawnBird>,
+    art: Res<Art>,
+) {
+    for ev in spawn_bird_events.iter() {
+        commands
+            .spawn_bundle(SpriteBundle {
+                sprite: Sprite {
+                    color: Color::rgb(randf(), randf(), randf()),
+                    custom_size: Some(Vec2::splat(BIRD_SIZE)),
+                    ..default()
+                },
+                texture: art.bird_icon.clone(),
+                transform: Transform::from_translation(ev.new_bird_pos.extend(randf() * 100.0)),
+                ..default()
+            })
+            .insert(Velocity {
+                velocity: ev.new_bird_velocity,
+            })
+            .insert(Gravity)
+            .insert(Bird)
+            .insert(Chaos)
+            .insert(DriftToCenter);
+    }
+}
+
+fn drift_to_center(
+    mut q: Query<(&mut Velocity, &Transform)>,
+    cam: Query<&Transform, With<Camera>>,
+    time: Res<Time>,
+) {
+    for (mut v, t) in q.iter_mut() {
+        v.velocity.x -= (t.translation.x - cam.single().translation.x)
+            * DRIFT_TO_CENTER_AMOUNT
+            * time.delta().as_secs_f32();
+    }
+}
+
+fn velocity(mut q: Query<(&Velocity, &mut Transform)>, time: Res<Time>) {
+    for (v, mut t) in q.iter_mut() {
+        t.translation += (v.velocity * time.delta().as_secs_f32()).extend(0.0);
+    }
+}
+
+fn gravity(mut q: Query<&mut Velocity, With<Gravity>>, time: Res<Time>) {
+    for mut v in q.iter_mut() {
+        v.velocity.y = v.velocity.y - time.delta().as_secs_f32() * GRAVITY;
+    }
+}
+
+fn bird_collision(
+    mut commands: Commands,
+    birds: Query<(Entity, &Sprite, &Transform), With<Bird>>,
+    obstacles: Query<(&Sprite, &Transform), With<Obstacle>>,
+) {
+    for (bird_entity, bird_sprite, bird_transform) in birds.iter() {
+        for (obstacle_sprite, obstacle_transform) in obstacles.iter() {
+            let collision_result = collide(
+                bird_transform.translation,
+                bird_sprite.custom_size.unwrap(),
+                obstacle_transform.translation,
+                obstacle_sprite.custom_size.unwrap(),
+            );
+            if collision_result.is_some() {
+                commands.entity(bird_entity).despawn();
+            }
+        }
+    }
+}
+
+fn bird_reproduction(
+    q: Query<(&Transform, &Velocity), With<Bird>>,
+    time: Res<Time>,
+    mut spawn_bird_events: EventWriter<SpawnBird>,
+) {
+    let mut bird_count = 0;
+    for (_t, _v) in q.iter() {
+        bird_count += 1;
+    }
+    for (t, v) in q.iter() {
+        if bird_count < MAX_BIRDS {
+            if randf() < BIRD_REPRODUCTION_CHANCE * time.delta().as_secs_f32() {
+                spawn_bird_events.send(SpawnBird {
+                    new_bird_pos: t.translation.truncate(),
+                    new_bird_velocity: v.velocity,
+                });
+            }
+        }
+    }
+}
+
+fn chaos(mut q: Query<&mut Velocity, With<Chaos>>, time: Res<Time>) {
```

Comment:
I feel like this could use a clearer name.

Reply:
brownian_drift?

Reply:
Perfect.